pkg/publish/docker.go

```diff
@@ -26,12 +26,21 @@ import (
 )
 
 // Docker publishes all images to the given hub
-func Docker(manifest model.Manifest, hub string, tags []string) error {
+func Docker(manifest model.Manifest, hub string, tags []string, cosignkey string) error {
 	dockerArchives, err := ioutil.ReadDir(path.Join(manifest.Directory, "docker"))
 	if err != nil {
 		return fmt.Errorf("failed to read docker output of release: %v", err)
 	}
 
+	// Only attempt to sign images if a valid cosign key is provided and we are
+	// able to run 'cosign public-key <key>'.
+	cosignEnabled := false
+	if cosignkey != "" {
+		if err := util.VerboseCommand("cosign", "public-key", "-key", cosignkey).Run(); err == nil {
+			cosignEnabled = true
+		}
```

Comment:
Let's print the error here. `cosign` key was passed, user expects it to be used.

Reply:
Is the point here to see the returned `public-key` or just to verify that the command exists? If the later, does it make sense to just do a `-version`?

Reply:
It does both. If a key is passed but the `cosign` binary is not present, it will abort attempting to sign the images without failure. If the cosign binary is present but the key is invalid or the permissions or wrong, it will also abort without a failure.

It wasn't clear to me if we wanted to return an error in that case and exit or if we should log a warning and continue.

Reply:
I'm fine logging a warning and continuing if `cosign`ing an image is an optional step. If we think the step is really required, then we should log and fail.

Reply:
Added a warning log.